tests/cases/compiler/unusedLocalsInRecursiveReturn.ts

```diff
@@ -0,0 +1,8 @@
+// @strict: true
+// @noEmit: true
+// @noUnusedLocals: true
+
+function recursive(arg: string, other: string) {

```

Comment:
this looks like it's testing something else

Reply:
In an earlier version, I was not unconditionally checking the return expression in `checkReturnStatement`. However, the expression should be checked even if we don't need to know its type, because it may contain errors and because we need to compute information for whether variables are used or not. The test shows the latter, and it's adapted from a break in the extended tests.